conductor_api/src/config.rs

```diff
@@ -147,13 +153,35 @@ impl Configuration {
                     instance.agent, instance.id
                 )
             })?;
-            self.dna_by_id(&instance.dna).is_some().ok_or_else(|| {
+            let dna_config = self.dna_by_id(&instance.dna);
+            dna_config.is_some().ok_or_else(|| {
                 format!(
                     "DNA configuration \"{}\" not found, mentioned in instance \"{}\"",
                     instance.dna, instance.id
                 )
             })?;
+            let dna_config = dna_config.unwrap();
+            let dna =
+                Arc::get_mut(&mut dna_loader).unwrap()(&PathBuf::from(dna_config.file.clone()))
+                    .map_err(|_| format!("Could not load DNA file \"{}\"", dna_config.file))?;
```

Comment:
get_mut will return a panic if they are Arc references to the same value. Is this what we want or can we switch to a mutex?

Reply:
So far we are fine since the dna_loader is only used during boot-up of the conductor, which happens in one thread.

But you might right that we should switch anyway. And I would not do that within the scope of this PR :)